distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -924,3 +919,16 @@ local_file_move (source : File) (destination : File) (replace_existing : Boolean
     File_Error.handle_java_exceptions source <|
         copy_options = if replace_existing then [StandardCopyOption.REPLACE_EXISTING.to_text] else []
         source.move_builtin destination copy_options
+
+# PRIVATE
+type File_System_SPI
+    private Entry protocol:Text typ
+
+    new protocol:Text typ = File_System_SPI.Entry protocol typ
+
+    private get_type protocol:Text _:Boolean =
+        vec = Self.get_types . filter (_.protocol == protocol)
+        if vec . not_empty then vec.first else Nothing
```

Comment:
I know that this is just a prototype but it changes the semantics from the `findSingleProvider` one.

Having two clashing implementations for the same protocol is a programming error of the library developers, so I think we should at least be getting a warning in such case.

Reply:
btw. wow I never knew that `Self` actually worked! Neat